builder/remotecontext/remote.go

```diff
@@ -110,7 +86,7 @@ func GetWithStatusError(address string) (resp *http.Response, err error) {
 //    - an io.Reader for the response body
 //    - an error value which will be non-nil either when something goes wrong while
 //      reading bytes from r or when the detected content-type is not acceptable.
-func inspectResponse(ct string, r io.Reader, clen int64) (string, io.ReadCloser, error) {
+func inspectResponse(ct string, r io.Reader, clen int64) (string, io.Reader, error) {
```

Comment:
Would it be cleaner for this function just to take a `ReadCloser` and return `ReadCloser` so caller doesn't need to do any wrapping?

Reply:
I guess you mean move the `ioutils.NewReadCloserWrapper()` into `inspectResponse()` ? I think either way it needs to be wrapped, it's just about where it happens.

That could work. It would mean that the two error returns in this function would have to call `r.Close()` instead of the one error case in the caller calling closer.

It also seems like it is more the responsibility of `downloadRemote()` since it is aware that it's a response body, where as `inspectResponse` only cares about a reader.

I have a very slight preference for the current implementation, but if you think it's cleaner to move it to inspectResponse I can do that.

I can also remove `closer()` that doesn't need to exist anymore, it can just be `response.Body.Close`

Reply:
ok, lets leave it like it is atm